src/server/actions/study.actions.ts

```diff
@@ -78,17 +85,20 @@ export const fetchStudiesForCurrentMemberAction = memberAction(async () => {
         .execute()
 })
 
-export const getStudyAction = memberAction(async (studyId) => {
-    const slug = await getOrgSlugFromActionContext()
+export const getStudyAction = userAction(async (studyId) => {
+    const ctx = await actionContext()
 
     return await db
         .selectFrom('study')
         .innerJoin('user', (join) => join.onRef('study.researcherId', '=', 'user.id'))
 
         // security, check user has access to record
-        .innerJoin('member', (join) =>
-            join.on('member.identifier', '=', slug).onRef('member.id', '=', 'study.memberId'),
+        .$if(Boolean(ctx?.orgSlug), (qb) =>
+            qb.innerJoin('member', (join) =>
+                join.on('member.identifier', '=', ctx.orgSlug!).onRef('member.id', '=', 'study.memberId'),
+            ),
         )
+        .$if(Boolean(ctx?.userId && !ctx?.orgSlug), (qb) => qb.where('study.researcherId', '=', ctx?.userId || ''))
 
```

Comment:
same comment as above we can simplify to https://github.com/safeinsights/management-app/blob/main/src/server/db/queries.ts#L38-L40

Reply:
much cleaner - thank you!